ddtelemetry/src/worker/mod.rs

```diff
@@ -154,7 +154,10 @@ impl TelemetryWorker {
         };
 
         // if no action is received, then it means the channel is stopped
-        action.unwrap_or(TelemetryActions::Lifecycle(LifecycleAction::Stop))
+        action.unwrap_or_else(|| {
+            self.config.restartable = false;
+            TelemetryActions::Lifecycle(LifecycleAction::Stop)
```

Comment:
Some documentation about how this `restartable` feature interacts with the rest of the `Lifecycle` of the `TelemetryWorker` would be good. This is not the only place where `LifecycleAction::Stop` is sent, and how do we know that we won't have `TelemetryWorker` instances staying around in some unexpected way.

Reply:
It's the only place where it's sent as a _definite_ termination event (i.e. where the handle has been dropped, causing the channel to be closed).

Reply:
Added comments :-)